src/core/flow.js

```diff
@@ -529,6 +529,19 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     // Reset activePlayers.
     ctx = { ...ctx, activePlayers: null };
 
+    // Remove player from playerOrder
+    if (arg && true === arg.remove) {
+      const playOrder = [...ctx.playOrder];
+
+      playOrder.splice(playOrder.indexOf(ctx.currentPlayer), 1);
```

Comment:
This can be more efficiently written as:
```
const playOrder = playOrder.filter(i => i != ctx.currentPlayer);
ctx = { ...ctx, playOrder };
```

Reply:
However, you should actually be using `playerID` (the player that called the event) rather than `currentPlayer`.  This value is passed around to all events and is available if you just extract it (similar to `EndStage` on line 563).